packages/core-data/src/entity-provider.js

```diff
@@ -128,18 +131,31 @@ export function useEntityProp( kind, type, prop ) {
  * @param {Object} [options.initialEdits]          Initial edits object for the entity record.
  * @param {string} [options.blocksProp='blocks']   The name of the entity prop that holds the blocks array.
  * @param {string} [options.contentProp='content'] The name of the entity prop that holds the serialized blocks.
+ * @param {string} [options.id]                    An entity ID to use instead of the context-provided one.
  *
  * @return {[WPBlock[], Function, Function]} The block array and setters.
  */
 export function useEntityBlockEditor(
```

Comment:
It doesn't seem to me like this should be in core-data or that it should be reusable.

Reply:
Definitely not in core-data.

Reply:
Why not?

I think you were the first one to suggest creating this hook.

Reply:
I don't remember suggesting this hook :P but definitely possible.

I think "block" is something specific to core-editor and not necessarily something an entity knows about, the entity just saves HTML in post content and not all entities have post content. Is this already shipped in WP 5.4 as stable?

Reply:
Yes

It's needed because the logic inside is nuanced and used in multiple places. It's also handy for plugin developers. Adding it to `block-editor` would make it depend on `core-data`, so I figured this was a better place since it should be ok for `core-data` to be aware of `block-editor`.

Reply:
`useEntityBlockEditor` was introduced in #19203, which was merged on Jan 8 and milestoned for Gutenberg 7.3.

Reply:
I agree core-data is better than block-editor. Seems like the ship has sailed :)

Reply:
Perhaps a common utility package could make sense too if things like this become common.

Reply:
I do there's a need for some WP specific utilities package. (a lot of the settings we pass to block-editor and tie it with APIs)

Reply:
Something lightweight to replace `editor` too.